src/EditorFeatures/CSharp/NavigationBar/CSharpEditorNavigationBarItemService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Extensibility.NavigationBar;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.NavigationBar;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.NavigationBar
+{
+    [ExportLanguageService(typeof(INavigationBarItemService), LanguageNames.CSharp), Shared]
+    internal class CSharpEditorNavigationBarItemService : AbstractEditorNavigationBarItemService
```

Comment:
this is all existing code.  it was just split from one service (that did both computing items as well as navigating to them) to two services.  The 'computing items' service is called INavigationBarItemService and moved down to the features layer.  The 'navigate to items' service is called IEditorNavigationBarService and stays at the editor layer.

Reply:
Because the editor version is already used by our IVT partners, we still support calling through it for both pieces of work.  However, teh C#/VB impl defers down to the feature layer to do the work (Which can now call into OOP to offload the computation).